### PR TITLE
Cats&Dogs model improvement

### PR DESCRIPTION
**Pull request by ermanok**
- Dataloader changed to augment data randomly at each epoch
- Training & QAT policies updated for optimized model training

### CODE REVIEW

File: datasets/cats_vs_dogs.py [hunk 1]
```diff
@@ -1,67 +1,84 @@
...
+        if d_type == 'train' and augment_data:
+            self.album_transform = album.Compose([
+                album.GaussNoise(var_limit=(1.0, 20.0), p=0.25),
+                album.RGBShift(r_shift_limit=15, g_shift_limit=15, b_shift_limit=15, p=0.5),
+                album.ColorJitter(p=0.5),
+                album.SmallestMaxSize(max_size=int(1.2*min(resize_size))),
+                album.ShiftScaleRotate(shift_limit=0.05, scale_limit=0.05, rotate_limit=15, p=0.5),
+                album.RandomCrop(height=resize_size[0], width=resize_size[1]),
+                album.HorizontalFlip(p=0.5),
+                album.Normalize(mean=(0.0, 0.0, 0.0), std=(1.0, 1.0, 1.0))])
```
Comment (MaximGorkem) [inline datasets/cats_vs_dogs.py:69]: I did not get the normalization with mean=0 and std=1. Doesn't this cause some pixel values to clamp since we apply ai8x.normalize on top?

File: policies/qat_policy_late_cd.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+start_epoch: 30
```
Comment (MaximGorkem) [inline policies/qat_policy_late_cd.yaml:2]: It is not very late, considering the 200 epochs. We might want to drop "late" from the filename.

**Review by MaximGorkem — COMMENTED** [review]
Looks good, there's only one point I could not quite get.

File: datasets/cats_vs_dogs.py (hunk 1 above)
Comment (MaximGorkem) [inline datasets/cats_vs_dogs.py:69]: One question: Doesn't this normalize all images & all channels independently? It is different than dividing the entire dataset by 255, which we employ in the other vision examples. Whichever correct way is needed to be applied to all examples.

File: policies/qat_policy_late_cd.yaml (hunk 2 above)
Comment (ermanok) [reply policies/qat_policy_late_cd.yaml]: Done!

**Review by rotx-maxim — APPROVED** [review]

File: datasets/cats_vs_dogs.py (hunk 1 above)
Comment (MaximGorkem) [reply datasets/cats_vs_dogs.py:71]: That's correct. I assumed that it divides each channel with its own maximum value. Doublechecked the documentation and it divides with max possible, which is 255.

File: scripts/train_catsdogs.sh [hunk 3]
```diff
@@ -1,2 +1,2 @@
 #!/bin/sh
-python train.py --epochs 250 --optimizer Adam --lr 0.001 --wd 0 --deterministic --compress policies/schedule-catsdogs.yaml --model ai85cdnet --dataset cats_vs_dogs --confusion --param-hist --embedding --device MAX78000 "$@"
+python train.py --epochs 200 --optimizer Adam --lr 0.001 --wd 0 --deterministic --compress policies/schedule-catsdogs.yaml --qat-policy policies/qat_policy_late_cd.yaml --model ai85cdnet --dataset cats_vs_dogs --confusion --param-hist --embedding --device MAX78000 "$@"
```
Comment (MaximGorkem) [inline scripts/train_catsdogs.sh:2]: Seems that "qat_policy_late_cd.yaml" should be replaced with "qat_policy_cd.yaml".

**Review by MaximGorkem — CHANGES_REQUESTED** [review]

File: scripts/train_catsdogs.sh (hunk 3 above)
Comment (ermanok) [reply scripts/train_catsdogs.sh]: Correct and fixed...

**Review by rotx-maxim — APPROVED** [review]
